File: app/schemas/analytics.py
from pydantic import BaseModel
from typing import Dict

class TasksAnalyticsResponse(BaseModel):
    total_tasks: int
    by_status: Dict[str, int]
    by_priority: Dict[str, int]
    by_assignee: Dict[str, int]
    status_percentage: Dict[str, float]
    priority_percentage: Dict[str, float]

File: client/src/types/analytics.ts
export interface TasksAnalyticsResponse {
  total_tasks: number
  by_status: Record<string, number>
  by_priority: Record<string, number>
  by_assignee: Record<string, number>
  status_percentage: Record<string, number>
  priority_percentage: Record<string, number>
}
